packages/apps/plugins/plugin-explorer/src/components/Tree/Tree.tsx

```diff
@@ -0,0 +1,108 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import React, { useEffect, useMemo, useState } from 'react';
+import { useResizeDetector } from 'react-resize-detector';
+
+import { type Space } from '@dxos/client/echo';
+import { createSvgContext, SVG, SVGContextProvider } from '@dxos/gem-core';
+
+import { HierarchicalEdgeBundling, RadialTree, TidyTree } from './layout';
+import { mapGraphToTreeData, type TreeNode } from './types';
+import { SpaceGraphModel } from '../Graph';
+
+// TODO(burdon): Create dge bundling graph using d3.hierarchy.
+// https://observablehq.com/@d3/hierarchical-edge-bundling?intent=fork
+
+type Renderer = (svg: SVGSVGElement, data: any, options: any) => void;
+
+export type LayoutType = 'tidy' | 'radial' | 'edge';
+
+// TODO(burdon): Normalize API and styling.
+
+export type TreeLayoutSlots = {
+  node?: string;
+  path?: string;
+  text?: string;
+};
```

Comment:
IMO we should deprecate the “slots” pattern in favor of either:
- supplying the downstream consumer with subcomponents that support `classNames`, or
- if `classNames` is the only supported prop on these subcomponents, just offering that as a prop name without calling anything “slots” (the latter implies the slot should/will implement a spread of all element props, but only `classNames` is supported making this a misnomer).

Reply:
I was struggling with this. I think there are Radix style components where the developer builds up the component where className is sufficient, but in this case where there's a more complex (hierarchical) layout I don't know how we'd do that. At the same time classNames seems to collide with the react-ui notion of providing a list (all for the given component). I also consider `classes` like material (i.e., for a map of classNames).

Reply:
> the react-ui notion of providing a list (all for the given component)

There is no such pattern that I can think of… certainly Table behaves differently, but both Tree and List export subcomponents. What are you referring to?

Reply:
At some point I'll want to redo this component to have drag-and-drop, and we might face some of the same challenges as the sidebar Tree (perhaps they will become more unified?)